src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java

```diff
@@ -130,11 +131,21 @@ public void run() {
                                     indexSettings.put("index.number_of_shards", 1);
                                     indexSettings.put("index.auto_expand_replicas", "0-all");
 
-                                    boolean ok = client.admin().indices().create(new CreateIndexRequest(opendistrosecurityIndex)
-                                            .settings(indexSettings))
-                                            .actionGet().isAcknowledged();
-                                    LOGGER.info("Index {} created?: {}", opendistrosecurityIndex, ok);
-                                    if(ok) {
+                                    boolean existsOrCreated = client.admin().indices()
+                                            .exists(new IndicesExistsRequest(opendistrosecurityIndex))
+                                            .actionGet().isExists();
+
+                                    if (existsOrCreated) {
+                                        LOGGER.info("Security index {} exists.", opendistrosecurityIndex);
+                                    } else {
+                                        LOGGER.info("Security index {} does not exist. Attempting to create new index.", opendistrosecurityIndex);
+                                        existsOrCreated = client.admin().indices().create(new CreateIndexRequest(opendistrosecurityIndex)
+                                                .settings(indexSettings))
+                                                .actionGet().isAcknowledged();
+                                        LOGGER.info("Index {} created?: {}", opendistrosecurityIndex, existsOrCreated);
+                                    }
+
+                                    if (existsOrCreated) {
```

Comment:
This code will be executed every time you restart an instance, which means that it will always override the existing config with default values, which is not what we want.  
Why not use instead some API to update the config values ?

Reply:
Yes, this will be executed on every restart. 

It will attempt to create config document with put-if-absent behavior by setting OpType to CREATE in ConfigHelper. 

If config exists, elastic throws a version conflict exception as specified here https://www.elastic.co/guide/en/elasticsearch/client/java-rest/master/java-rest-high-document-index.html#java-rest-high-document-index-response so plugin will simply skip config documents that already exist.

Reply:
How this will help you to load "audit.yml" ? Won't this simply throw an exception during first attempt to put the "config.yml" file ?

Reply:
Say the existing cluster has documents roles, rolemapping, internalusers ... already loaded from their yml files when the cluster first started. 

Now you want to include audit configuration from audit.yml file. 

With this change, you include the audit.yml file and restart the cluster. 
Updates will fail for roles, rolemapping, internalusers .. (throwing exception) but will pass for audit as the document does not exist in the index yet (Put-If-Absent) behavior.